test/combinations.jl

```diff
@@ -1,429 +1,365 @@
-# This section tests for:
-# - All supported combinations of integral(f, ::Geometry, ::IntegrationAlgorithm) produce accurate results
-# - Invalid applications of integral aliases (e.g. lineintegral) produce a descriptive error
+"""
+This file includes tests for:
+- Supported combinations {::Geometry, ::IntegrationRule} for `integral(f, geometry, rule)`.
+- Integrand functions returning a `Unitful.Quantity` or a `Vector{Unitful.Quantity}`.
+- Callable objects in place of an integrand function.
+- Unsupported combinations produce a descriptive and useful error message.
+- The appropriate alias function, e.g. `lineintegral` for 1D geometries, works.
+- Invalid applications of integral aliases produce a descriptive and useful error message.
+- (Planned) Usage of non-default DifferentiationMethods.
+"""
+
+#===============================================================================
+                        Test Generation Infrastructure
+===============================================================================#
+
+@testsnippet Combinations begin
+    using LinearAlgebra: norm
+    using Meshes
+    using MeshIntegrals
+    using Unitful
+
+    # Used for testing callable objects as integrand functions
+    struct Callable{F <: Function}
+        f::F
+    end
+    (c::Callable)(p::Meshes.Point) = c.f(p)
+
+    # Stores a testable combination
+    struct TestableGeometry{F <: Function, G <: Geometry, U <: Unitful.Quantity}
+        integrand::F
+        geometry::G
+        solution::U
+    end
+
+    # Used to indicate which features are supported for a particular geometry
+    struct SupportStatus
+        # Alias Functions
+        lineintegral::Bool
+        surfaceintegral::Bool
+        volumeintegral::Bool
+        # IntegrationRules
+        gausskronrod::Bool
+        gausslegendre::Bool
+        hadaptivecubature::Bool
+        # DifferentiationMethods
+        # autoenzyme::Bool
+    end
+
+    # Shortcut constructor for geometries with typical support structure
+    function SupportStatus(sym::Symbol)
+        if sym == :line
+            aliases = Bool.((1, 0, 0))
+            rules = Bool.((1, 1, 1))
+            return SupportStatus(aliases..., rules...)
+        elseif sym == :surface
+            aliases = Bool.((0, 1, 0))
+            rules = Bool.((1, 1, 1))
+            return SupportStatus(aliases..., rules...)
+        elseif sym == :volume
+            aliases = Bool.((0, 0, 1))
+            rules = Bool.((0, 1, 1))
+            return SupportStatus(aliases..., rules...)
+        else
+            error("Unrecognized SupportStatus shortcut $(string(sym))")
+        end
+    end
```

Comment:
I think this is a good idea. When I first implemented this I expected there to be more outliers that didn’t conform to the standard support structure, but after the recent `_ParametricGeometry` PR’s it looks like support is currently purely a function of `paramdim`s, which is really nice. I’ll try to get all of the tests converted when I get back to my computer.

Reply:
I forgot to go back and add a question about Box-4D. It’s currently on extended/local-only testing because it’s niche and CI times are pretty lengthy. In terms of implementation for `Box` methods, there’s actually nothing special about 2D vs 4D vs 32D, etc. Should we bother keeping that test set?

Reply:
I agree that there's no fundamental difference between boxes of different dimensions, but I also don't see a big downside of having one test with a geometry of paramdim > 3 and putting it in the extended testing. So if possible I would keep it just to be sure that everything works dimension-independent.

Reply:
I just restored the 4D test.